src/components/diagrams/diagram-pane.js

```diff
@@ -409,27 +704,55 @@ export function DiagramPane({
      * FORCED UPDATE MECHANISM
      */
 
-    // Updates a particular SLD diagram or every diagram for the current node
-    const updateDiagram = useCallback((id) => {
-        if (id) {
-            viewsRef.current
-                .find(
-                    (diagramView) =>
-                        diagramView.id === id &&
-                        diagramView.svgType !== DiagramType.NETWORK_AREA_DIAGRAM
-                )
-                ?.ref?.current?.reloadSvg();
-        } else {
-            viewsRef.current.forEach((diagramView) => {
-                // We search an instance of the current node's ID inside the diagram's URL to determine if the diagram should be updated
-                if (
-                    diagramView.svgUrl &&
-                    diagramView.svgUrl.indexOf(currentNodeRef.current?.id) !==
-                        -1
-                ) {
-                    diagramView.ref?.current?.reloadSvg();
+    // Updates particular diagrams or every diagram for the current node
+    const updateDiagrams = useCallback((ids) => {
```

Comment:
If this function is rewriten, maybe that's a good time to change it so it is clearer that updateDiagrams(undefined) will update all diagrams, whereas updateDiagrams(something) will only update something. I would use two different functions to clarify this.

Reply:
I kept the previous design but I agree with you